CMSgov/cmcs-eregulations
# EREGCSC-1296 - Add Supplemental Content search to JS1 Resources prototype page

Resolves [EREGCSC-1296](https://jiraent.cms.gov/browse/EREGCSC-1296)

**Description**
Adds Supplemental Content search to JS1 Resources prototype page.

**This pull request changes**
* Updates Django class `SupplementalContentView` to accept `q` parameter to further filter supplemental content by that parameter, in addition to subpart/section/etc.
* Allows Django class `SupplementalContentView` to accept `all` string for `title` and `part`.  This way, supplying `all` for `title` and `part` while also supplying `q` string will return supplemental content for all parts that contains 'q' query string.
* Adds "real" related section links to every piece of supplemental content that is returned.  Links go to prototype's Part view, but does not scroll down to section.  This mirrors the current functionality of the [PDResources](https://d164t2gueqvg28.cloudfront.net/pdresources) "Relevant Regulations" links in the right sidebar of search results.

**Steps to manually verify this change**

1. Go to [prototype deployment for this PR](https://d164t2gueqvg28.cloudfront.net/resources)
2. use dropdowns to filter resources by section/etc and compare to [main branch prototype in dev environment](https://d3hdo1gedk3k52.cloudfront.net/) to make sure they still function the same
3. Start using the search text box in [prototype deployment for this PR](https://d164t2gueqvg28.cloudfront.net/resources) and search for random strings with and without additional filters from the dropdowns
